aries_cloudagent/utils/profiles.py

```diff
@@ -29,3 +34,26 @@ def subwallet_type_not_same_as_base_wallet_raise_web_exception(
         raise web.HTTPForbidden(
             reason="Subwallet type must be the same as the base wallet type"
         )
+
+
+async def get_subwallet_profiles_from_storage(root_profile: Profile) -> list[Profile]:
+    """Get subwallet profiles from storage."""
+    subwallet_profiles = []
+    base_storage_search = root_profile.inject(BaseStorageSearch)
+    search_session = base_storage_search.search_records(
+        type_filter=WalletRecord.RECORD_TYPE, page_size=10
+    )
+    while search_session._done is False:
+        wallet_storage_records = await search_session.fetch()
+        for wallet_storage_record in wallet_storage_records:
+            wallet_record = WalletRecord.from_storage(
+                wallet_storage_record.id,
+                json.loads(wallet_storage_record.value),
+            )
+            subwallet_profiles.append(
+                await MultitenantManager(root_profile).get_wallet_profile(
+                    base_context=root_profile.context,
+                    wallet_record=wallet_record,
+                )
+            )
+    return subwallet_profiles
```

Comment:
Hi @jamshale - we're experiencing some issues with the `check_for_wallet_upgrades_in_progress` method.
This method gets called upon startup, and if we have existing wallets in the DB, then we get the following stacktrace:

```sh
Traceback (most recent call last):
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/askar/store.py\", line 154, in open_store
    store = await Store.open(
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_askar/store.py\", line 352, in open
    return Store(await bindings.store_open(uri, key_method, pass_key, profile), uri)
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_askar/bindings/__init__.py\", line 84, in store_open
    return await invoke_async(
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_askar/bindings/lib.py\", line 362, in invoke_async
    return await self.loaded.invoke_async(
  File \"/usr/local/lib/python3.9/asyncio/futures.py\", line 284, in __await__
    yield self  # This tells Task to wait for completion.
  File \"/usr/local/lib/python3.9/asyncio/tasks.py\", line 328, in __wakeup
    future.result()
  File \"/usr/local/lib/python3.9/asyncio/futures.py\", line 201, in result
    raise self._exception
aries_askar.error.AskarError: Error fetching store configuration
Caused by: error returned from database: Request returned an error: database \"472436df-6126-4083-85fb-1eacc0ecfcf5\" does not exist.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/commands/start.py\", line 72, in init
    await startup
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/commands/start.py\", line 29, in start_app
    await conductor.start()
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/core/conductor.py\", line 522, in start
    await self.check_for_wallet_upgrades_in_progress()
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/core/conductor.py\", line 846, in check_for_wallet_upgrades_in_progress
    subwallet_profiles = await get_subwallet_profiles_from_storage(
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/utils/profiles.py\", line 54, in get_subwallet_profiles_from_storage
    await MultitenantManager(root_profile).get_wallet_profile(
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/multitenant/manager.py\", line 85, in get_wallet_profile
    profile, _ = await wallet_config(context, provision=provision)
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/config/wallet.py\", line 54, in wallet_config
    profile = await mgr.open(context, profile_cfg)
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/askar/profile.py\", line 300, in open
    opened = await store_config.open_store(provision=False)
  File \"/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/askar/store.py\", line 168, in open_store
    raise ProfileError(\"Error opening store\") from err
aries_cloudagent.core.error.ProfileError: Error opening store"
```

I don't quite understand what could be going wrong, because that error:
`Request returned an error: database \"472436df-6126-4083-85fb-1eacc0ecfcf5\" does not exist.`
is referring to a wallet_id that does exist in the DB. Maybe it's just trying to read it incorrectly? 

We're using askar wallet_type, and postgres_storage.

Does any of this look familiar to you, or do you have any idea what might be going wrong? Any help will be appreciated!

Reply:
Awesome, thanks for the feedback!

One thing that may make our situation unique is the use of Amazon RDS Proxy, for the DB.

I'm personally not so informed on our infra config, but our SRE mentioned that there's a quirk with bootstrapping the DB. We modify the `ACAPY_WALLET_STORAGE_CONFIG` after an initialisation, to point to an RDS Proxy URL.

I'll need to gather more details to correctly relay the specifics... but that's one factor that may make our setup different.

It sounds like provision=True may be necessary in this case? If you think that might be it, I could try modify that on a fork and test on our end. But only by tomorrow 👌